packages/server/src/sdk/app/rows/search/filters.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  FieldType,
+  RelationshipFieldMetadata,
+  SearchFilters,
+  Table,
+} from "@budibase/types"
+
+export function getRelationshipColumns(table: Table): {
+  name: string
+  definition: RelationshipFieldMetadata
+}[] {
+  return Object.entries(table.schema)
+    .filter(entry => entry[1].type === FieldType.LINK)
+    .map(entry => ({
+      name: entry[0],
+      definition: entry[1] as RelationshipFieldMetadata,
+    }))
+}
+
+export function getTableIDList(
+  tables: Table[]
+): { name: string; id: string }[] {
+  return tables
+    .filter(table => table.originalName)
+    .map(table => ({ id: table._id!, name: table.originalName! }))
+}
+
+export function updateFilterKeys(
+  filters: SearchFilters,
+  updates: { original: string; updated: string }[]
+): SearchFilters {
+  const makeFilterKeyRegex = (str: string) =>
+    new RegExp(`^${str}\\.|:${str}\\.`)
+  for (let filter of Object.values(filters)) {
+    if (typeof filter !== "object") {
```

Comment:
I don't think this is quite doing what you're hoping it to do. I fell over this myself recently, for example:

```javascript
> typeof null
'object'
> typeof []
'object'
```

I _think_ you want https://lodash.com/docs/4.17.15#isPlainObject because even `_.isObject` returns true for arrays.